python/cuml/feature_extraction/text.py

```diff
@@ -0,0 +1,603 @@
+                 strip_accents=None, lowercase=True, preprocessor=None,
+                 tokenizer=None, stop_words=None, token_pattern=None,
+                 ngram_range=(1, 1), analyzer='word', max_df=1.0, min_df=1,
+                 max_features=None, vocabulary=None, binary=False,
+                 dtype=cp.float32, delimiter=' '):
+        self.preprocessor = preprocessor
+        self.analyzer = analyzer
+        self.lowercase = lowercase
+        self.stop_words = stop_words
+        self.max_df = max_df
+        self.min_df = min_df
+        if max_df < 0 or min_df < 0:
+            raise ValueError("negative value for max_df or min_df")
+        self.max_features = max_features
+        if max_features is not None:
+            if not isinstance(max_features, int) or max_features <= 0:
+                raise ValueError(
+                    "max_features=%r, neither a positive integer nor None"
+                    % max_features)
+        self.ngram_range = ngram_range
+        self.vocabulary = vocabulary
+        self.binary = binary
+        self.dtype = dtype
+        self.delimiter = delimiter
+        if dtype not in CUPY_SPARSE_DTYPES:
+            msg = f"Expected dtype in {CUPY_SPARSE_DTYPES}, got {dtype}"
+            raise ValueError(msg)
+
+        sklearn_params = {"input": input,
+                          "encoding": encoding,
+                          "decode_error": decode_error,
+                          "strip_accents": strip_accents,
+                          "tokenizer": tokenizer,
+                          "token_pattern": token_pattern}
+        self._check_sklearn_params(analyzer, sklearn_params)
+
+    def _count_vocab(self, tokenized_df):
+        """Count occurrences of tokens in each document."""
+        # Transform string tokens into token indexes from 0 to len(vocab)
+        # The indexes are based on lexicographical ordering.
+        tokenized_df['token'] = tokenized_df['token'].astype('category')
+        tokenized_df['token'] = tokenized_df['token'].cat.set_categories(
+            self.vocabulary_
+        )._column.codes
+
+        # Count of each token in each document
+        count_df = (
+            tokenized_df[["doc_id", "token"]]
+            .groupby(["doc_id", "token"])
+            .size()
+            .reset_index()
+            .rename({0: "count"})
+        )
+
+        return count_df
+
+    def _filter_and_renumber(self, df, keep_values, column):
+        """Filter dataframe to keep only values from column matching
+        keep_values."""
+        df[column] = (
+            df[column].astype('category')
+            .cat.set_categories(keep_values)
+            ._column.codes
+        )
+        df = df.dropna(subset=column)
+        return df
+
+    def _limit_features(self, count_df, vocab, high, low, limit):
+        """Remove too rare or too common features.
+
+        Prune features that are non zero in more samples than high or less
```

Comment:
Docstring should clarify that this sets self.vocabulary_ and self.stop_words_ whenever we modify self. in a way that might be non-obvious, it's nice to be clear

Reply:
Good point. Clarifying this.